src/chainrules.jl

```diff
@@ -1,123 +1,43 @@
-    function ifft_pullback(ȳ)
-        x̄ = project_x(invN .* fft(ChainRulesCore.unthunk(ȳ), dims))
-        return ChainRulesCore.NoTangent(), x̄, ChainRulesCore.NoTangent()
-    end
-    return y, ifft_pullback
-end
-
-function ChainRulesCore.frule((_, Δx, _, _), ::typeof(irfft), x::AbstractArray, d::Int, dims)
-    y = irfft(x, d, dims)
-    Δy = irfft(Δx, d, dims)
-    return y, Δy
-end
-function ChainRulesCore.rrule(::typeof(irfft), x::AbstractArray, d::Int, dims)
-    y = irfft(x, d, dims)
-
-    # compute scaling factors
-    halfdim = first(dims)
-    n = size(x, halfdim)
-    invN = normalization(y, dims)
-    twoinvN = 2 * invN
-    scale = reshape(
-        [i == 1 || (i == n && 2 * (i - 1) == d) ? invN : twoinvN for i in 1:n],
-        ntuple(i -> i == first(dims) ? n : 1, Val(ndims(x))),
-    )
-
-    project_x = ChainRulesCore.ProjectTo(x)
-    function irfft_pullback(ȳ)
-        x̄ = project_x(scale .* rfft(real.(ChainRulesCore.unthunk(ȳ)), dims))
-        return ChainRulesCore.NoTangent(), x̄, ChainRulesCore.NoTangent(), ChainRulesCore.NoTangent()
+for f in (:fft, :bfft, :ifft, :rfft)
+    pf = Symbol("plan_", f)
+    @eval begin
+        function ChainRulesCore.frule((_, Δx, _), ::typeof($f), x::AbstractArray, dims) 
+            y = $f(x, dims)
+            Δy = $f(Δx, dims)
+            return y, Δy
+        end
+        function ChainRulesCore.rrule(::typeof($f), x::T, dims) where {T<:AbstractArray}
+            y = $f(x, dims)
+            project_x = ChainRulesCore.ProjectTo(x)
+            ax = axes(x)
+            function fft_pullback(ȳ)
+                x̄ = project_x($pf(similar(T, ax), dims)' * ChainRulesCore.unthunk(ȳ))
+                return ChainRulesCore.NoTangent(), x̄, ChainRulesCore.NoTangent()
+            end
+            return y, fft_pullback
+        end
     end
-    return y, irfft_pullback
 end
 
-function ChainRulesCore.frule((_, Δx, _), ::typeof(bfft), x::AbstractArray, dims)
-    y = bfft(x, dims)
-    Δy = bfft(Δx, dims)
-    return y, Δy
-end
-function ChainRulesCore.rrule(::typeof(bfft), x::AbstractArray, dims)
-    y = bfft(x, dims)
-    project_x = ChainRulesCore.ProjectTo(x)
-    function bfft_pullback(ȳ)
-        x̄ = project_x(fft(ChainRulesCore.unthunk(ȳ), dims))
-        return ChainRulesCore.NoTangent(), x̄, ChainRulesCore.NoTangent()
+for f in (:brfft, :irfft)
+    pf = Symbol("plan_", f)
+    @eval begin
+        function ChainRulesCore.frule((_, Δx, _), ::typeof($f), x::AbstractArray, d::Int, dims) 
+            y = $f(x, d::Int, dims)
+            Δy = $f(Δx, d::Int, dims)
+            return y, Δy
+        end
+        function ChainRulesCore.rrule(::typeof($f), x::T, d::Int, dims) where {T<:AbstractArray}
+            y = $f(x, d, dims)
+            project_x = ChainRulesCore.ProjectTo(x)
+            ax = axes(x)
+            function fft_pullback(ȳ)
+                x̄ = project_x($pf(similar(T, ax), d, dims)' * ChainRulesCore.unthunk(ȳ))
```

Comment:
Same here, I think the rules for `brfft` etc. should not have to deal with plans.